### PR TITLE
DeepSpeed integration tests for Lite

### PR DESCRIPTION
**Pull request by awaelchli**
## What does this PR do?

Similar procedure as to #14815, covering the deepspeed strategy. 

- adopted: This test was fully or partially adopted into Lite
- skipped: The test is legacy or questionable whether it is of any use at all
- unimplemented: This test covers functionality only available in Trainer, and can't be ported to Lite. Example: Gradient accumulation magic

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
I made sure I had fun coding 🙃

cc @borda @awaelchli @rohitgr7 @akihironitta @carmocca @justusschock

### CODE REVIEW

File: tests/tests_lite/test_lite.py [hunk 1]
```diff
@@ -452,79 +451,3 @@ def test_autocast():
     with lite.autocast():
         lite._precision_plugin.forward_context().__enter__.assert_called()
     lite._precision_plugin.forward_context().__exit__.assert_called()
-
-
-@RunIf(min_cuda_gpus=2, standalone=True, deepspeed=True)
-def test_deepspeed_multiple_models():
```
Comment (awaelchli) [inline tests/tests_lite/test_lite.py:458]: moved this test to the integration file. no changes otherwise.

File: src/pytorch_lightning/strategies/deepspeed.py [hunk 2]
```diff
@@ -654,23 +655,30 @@ def _auto_select_batch_size(self) -> int:
         # train_micro_batch_size_per_gpu is used for throughput logging purposes
         # by default we try to use the batch size of the loader
         assert self.lightning_module is not None
-        batch_size = 1
```
Comment (awaelchli) [inline src/pytorch_lightning/strategies/deepspeed.py:657]: refactored so that I could use the shared `_get_dataloader_batch_size` function. Retrospectively, maybe not worth it. meh :(

File: tests/tests_pytorch/strategies/test_deepspeed_strategy.py [hunk 3]
```diff
@@ -120,6 +120,7 @@ def deepspeed_zero_config(deepspeed_config):
     return {**deepspeed_config, "zero_allow_untested_optimizer": True, "zero_optimization": {"stage": 2}}
 
 
+# lite: adopted
```
Comment (awaelchli) [inline tests/tests_pytorch/strategies/test_deepspeed_strategy.py:123]: todo: remove notes before merge

File: src/lightning_lite/strategies/deepspeed.py [hunk 4]
```diff
@@ -559,13 +563,14 @@ def _create_default_config(
                     "max_in_cpu": max_in_cpu,
                     "pin_memory": pin_memory,
                 }
-            cfg = {
```
Comment (awaelchli) [inline src/lightning_lite/strategies/deepspeed.py:562]: only cosmetic changes

File: src/lightning_lite/strategies/deepspeed.py [hunk 5]
```diff
@@ -186,8 +186,6 @@ def __init__(
 
             logging_batch_size_per_gpu: Config used in DeepSpeed to calculate verbose timing for logging
                 on a per sample per second basis (only displayed if logging=logging.INFO).
-                If set to "auto", the plugin tries to infer this from
-                the train DataLoader's BatchSampler, else defaults to 1.
```
Comment (awaelchli) [inline src/lightning_lite/strategies/deepspeed.py:190]: We can't infer this automatically at the moment. See note about limitation below

File: src/lightning_lite/strategies/strategy.py [hunk 6]
```diff
@@ -240,6 +240,7 @@ def save_checkpoint(
 
     def get_module_state_dict(self, module: Module) -> Dict[str, Union[Any, Tensor]]:
         """Returns model state."""
+        # TODO(lite): Integrate this into Lightning Lite
```
Comment (carmocca) [inline src/lightning_lite/strategies/strategy.py:243]: What does this mean? It's in lightning lite already

**Review by carmocca — APPROVED** [review]

File: src/lightning_lite/strategies/strategy.py (hunk 6 above)
Comment (awaelchli) [reply src/lightning_lite/strategies/strategy.py:243]: Yes, the api is here but it would be nice if we could call it in Lite, or expose it in some way. For example: #14816

**Review by justusschock — APPROVED** [review]

**Review by tchaton — APPROVED** [review]
LGTM !
